From 4677dd55d55832dab1e4f41fe9a7fabe8bb95197 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Thu, 24 Nov 2016 08:27:45 +0100
Subject: [PATCH] ivshmem-net: Map shmem region as RAM

No need for special caching, simply map the shared memory region like
RAM, thus write-back. This gives us another order of magnitude in
throughput.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
(cherry picked from commit af59c6541a65622cab498851e01653dd378cd9f8)
---
 drivers/net/ivshmem-net.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ivshmem-net.c
+++ b/drivers/net/ivshmem-net.c
@@ -802,7 +802,7 @@ static int ivshm_net_probe(struct pci_de
 	if (!devm_request_mem_region(&pdev->dev, shmaddr, shmlen, DRV_NAME))
 		return -EBUSY;
 
-	shm = devm_memremap(&pdev->dev, shmaddr, shmlen, MEMREMAP_WC);
+	shm = devm_memremap(&pdev->dev, shmaddr, shmlen, MEMREMAP_WB);
 	if (!shm)
 		return -ENOMEM;
 
